docs/src/tutorial/particles.jl

```diff
@@ -0,0 +1,172 @@
+# # Particles and Phase Space Points
+
+# There are three layers of abstraction from particles to phase space points in the QEDjl project:
+# - [`AbstractParticleType`](@extref QEDbase.AbstractParticleType): Base type for singleton particle type definitions. We also call these *species*.
+# - [`AbstractParticleStateful`](@extref QEDbase.AbstractParticleStateful): Base type for particles with a direction and carrying a momentum.
+# - [`AbstractPhaseSpacePoint`](@extref QEDbase.AbstractPhaseSpacePoint): Representation of a point in the phase space for a combination of an [`AbstractProcessDefinition`](@extref QEDbase.AbstractProcessDefinition), [`AbstractModelDefinition`](@extref QEDbase.AbstractModelDefinition), and [`AbstractPhasespaceDefinition`](@extref QEDbase.AbstractPhasespaceDefinition).
+
+# This manual is intended to showcase the basic usage of these types and their implementations in QEDcore.
+
+using QEDcore
+
+# To use concrete process definitions and models, we also need to use [`QEDprocesses.jl`](https://github.com/QEDjl-project/QEDprocesses.jl)
+
+using QEDprocesses
+
+# ## Particle Types
+
+# QEDcore currently defines the three basic particle types of QED, [`Electron`](@ref), [`Positron`](@ref), and [`Photon`](@ref), and a type hierarchy for them:
+
+@assert Photon <: MajoranaBoson
+@assert Electron <: Fermion
+@assert Positron <: AntiFermion
+
+# All of these are subtypes of [`AbstractParticleType`](@extref QEDbase.AbstractParticleType).
+# There are also convenience functions in Julia convention:
+
+@assert is_boson(Photon())
+@assert is_particle(Electron())
+@assert is_anti_particle(Positron())
+
+@assert !is_boson(Electron())
+@assert !is_anti_particle(Electron())
+@assert !is_fermion(Photon())
+
+# These functions are part of QEDbase.jl's (TODO: enable link once QEDbase docs are up)`[particle interface](@extref QEDbase particles)`.
+
+# ## ParticleStateful
+
+# [`ParticleStateful`](@ref) is the implementation of QEDbase's [`AbstractParticleStateful`](@extref QEDbase.AbstractParticleStateful) interface. It represents a particle with a [`direction`](@extref QEDbase.ParticleDirection) (as used in the context of scattering processes, [`Incoming`](@extref QEDbase.Incoming), [`Outgoing`](@extref QEDbase.Outgoing), or [`UnknownDirection`](@extref QEDbase.UnknownDirection)), a particle species ([`Electron`](@ref), [`Positron`](@ref), [`Photon`](@ref), ...), and a 4-momentum Vector.
```

Comment:
I think it is more explicit if you use the package name where a type comes from directly, e.g. ```[`QEDbase.Incoming`](@extref)``` instead of ```[`Incoming`](@extref QEDbase.Incoming)```